### ISSUE TITLE
Unexpected behaviour when parsing a proof

### ISSUE THREAD

**Issue by asr** [msakai/Smtlib#1]
Using the version of Smtlib from the master branch of https://github.com/MfesGA/Smtlib (commit https://github.com/MfesGA/Smtlib/commit/e899a2789737c6f4afe5326cd6b9f58918da6a02), I got the correct result:

```
$ ./Test id.z3-proof 
SexprSxp [SexprSxp [SexprSymbol "set-logic",SexprSymbol "QF_UF"],SexprSxp [SexprSymbol "proof",SexprSxp [SexprSymbol "let",SexprSxp [SexprSxp [SexprSymbol "@x35",SexprSxp [SexprSymbol "monotonicity",SexprSxp [SexprSymbol "rewrite",SexprSxp [SexprSymbol "=",SexprSxp [SexprSymbol "=>",SexprSymbol "p",SexprSymbol "p"],SexprSymbol "true"]],SexprSxp [SexprSymbol "=",SexprSxp [SexprSymbol "not",SexprSxp [SexprSymbol "=>",SexprSymbol "p",SexprSymbol "p"]],SexprSxp [SexprSymbol "not",SexprSymbol "true"]]]]],SexprSxp [SexprSymbol "let",SexprSxp [SexprSxp [SexprSymbol "@x39",SexprSxp [SexprSymbol "trans",SexprSymbol "@x35",SexprSxp [SexprSymbol "rewrite",SexprSxp [SexprSymbol "=",SexprSxp [SexprSymbol "not",SexprSymbol "true"],SexprSymbol "false"]],SexprSxp [SexprSymbol "=",SexprSxp [SexprSymbol "not",SexprSxp [SexprSymbol "=>",SexprSymbol "p",SexprSymbol "p"]],SexprSymbol "false"]]]],SexprSxp [SexprSymbol "mp",SexprSxp [SexprSymbol "asserted",SexprSxp [SexprSymbol "not",SexprSxp [SexprSymbol "=>",SexprSymbol "p",SexprSymbol "p"]]],SexprSymbol "@x39",SexprSymbol "false"]]]]]
```

but using the version in your develop branch (commit 6aabc07), I got the wrong result

``` bash
$ ./Test id.z3-proof 
error parsing the proof (left)
```

Have you any idea why I'm getting the above result using the version in your develop branch?

I attach the files [Test.hs.txt](https://github.com/msakai/Smtlib/files/288119/Test.hs.txt) and [id.z3-proof.txt](https://github.com/msakai/Smtlib/files/288120/id.z3-proof.txt).

**Comment by msakai** [comment]
Thank you for letting me know that.

I think it is the problem of SMT-LIB2 standard.

It says

> A ⟨symbol⟩ is either a simple symbol or a quoted symbol.
> A simple symbol is any non-empty sequence of elements of ⟨letter⟩ and ⟨digit⟩ and the characters
> ~!@$%^&*_-+=<>.?/
> that does not start with a digit and is not a reserved word.
> …

and define ⟨s_expr⟩ and ⟨get_proof_response⟩  as follows.

> ⟨s_expr⟩ ::= ⟨spec_constant⟩ | ⟨symbol⟩ | ⟨keyword⟩ | ( ⟨s_expr⟩∗ )
> ⟨get_proof_response⟩ ::= ⟨s_expr⟩

I implemented the syntax as it is, but then ⟨get_proof_response⟩  cannot contain `set-logic`, `let`, etc. as simple symbols.

I fixed the problem in 832042e by allowing reserved words to be used as simple symbols in ⟨s_expr⟩ category.

**Comment by asr** [comment]
Thanks for fixing the issue.
